### ISSUE TITLE
Split DNS for Corporate DNS Servers

### ISSUE THREAD

**Issue by strawgate** [tailscale/tailscale#953]
**Is your feature request related to a problem? Please describe.**

When DNS Servers are provided all DNS queries go to the specified DNS server. If the DNS server is unreachable due to it being offline, a tailscale ACL, or some other reason, the user is generally unable to access the internet or any other resources (or at least they experience long delays).

**Describe the solution you'd like**

It would appear that when Magic DNS is enabled, the DNS for the tunnel is set to the Magic DNS address and then Magic DNS forwards to the specified corporate DNS Servers.

It would be nice if we could define a set of domains that our corporate DNS servers are authoritative for so that magic DNS doesn't need to reach out to corporate DNS to resolve domain names on the general internet. For instance *.mydomain.com uses my DNS servers but everything else uses the system's default DNS.

In other words, if I specify a set of DNS servers and there is some issue with corporate access, I dont want to impact the user's ability to access the internet.

**Comment by DentonGentry** [comment]
Updated to use terminology we're using elsewhere about this.

**Comment by danderson** [comment]
Some notes as I research how to make this happen: per-domain resolvers in Windows is set through the Name Resolution Policy Table, documented at https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-gpnrpt/8cc31cb9-20cb-4140-9e85-3e08703b4745 . It boils down to a set of registry keys to define a rule matching on particular DNS suffixes, which override the usual per-interface DNS configurations (which are used to determine the "default route" for DNS queries).

I have other notes in the design doc figuring out an overall plan for cross-platform DNS, hopefully I can share that as well at some point. But that piece of MS documentation was tricky to track down, so making a note now.

**Comment by danderson** [comment]
Update: split DNS support is implemented and works on Linux, Windows, macOS and iOS. Android support is still pending, as is the admin panel UI change to expose split functionality.

**Comment by rosszurowski** [comment]
With v1.8 now released (public announcement coming early next week), we're just finalizing the admin panel UI for managing split DNS.

There's a few tricky pieces here. If anyone wanting to use a split DNS config with Tailscale is available for a usability test of this new UI, drop me a note at ross@tailscale.com. Happy to send some Tailscale swag as a thank you for the time :)

**Comment by rosszurowski** [comment]
We've now launched [the new UI for managing Split DNS](https://login.tailscale.com/admin/dns), and will be updating our knowledgebase with details later today. Anyone waiting for this feature can upgrade to v1.8 and set up their config in the admin panel today. Please let us know how it works for you!
